package.json

```diff
@@ -25,29 +25,40 @@
     "@wordpress/blocks": "6.21.0",
     "@wordpress/components": "9.4.1",
     "@wordpress/compose": "3.25.0",
+    "@wordpress/core-data": "2.26.0",
     "@wordpress/data": "4.22.3",
     "@wordpress/data-controls": "1.16.3",
     "@wordpress/dom-ready": "2.11.0",
     "@wordpress/edit-post": "3.27.0",
+    "@wordpress/editor": "9.26.0",
     "@wordpress/element": "2.16.0",
+    "@wordpress/escape-html": "1.9.0",
+    "@wordpress/hooks": "2.9.0",
     "@wordpress/i18n": "3.14.0",
     "@wordpress/icons": "2.10.0",
     "@wordpress/keycodes": "2.19.2",
+    "@wordpress/plugins": "2.25.0",
+    "@wordpress/token-list": "1.12.0",
     "@wordpress/url": "2.17.0",
     "classnames": "2.3.1",
     "interpolate-components": "1.1.1",
+    "jquery": "3.5.1",
     "lodash": "4.17.21",
+    "react": "16.11.0",
```

Comment:
For `react`, there are at least one of our dev dependencies that conflicted with the version installed (`16.11.0`). I forced the install because that was what we're using anyway according to `npm-check`, but we might want to take a look at that.

Reply:
Another thing that we might want to consider is to get rid of the `react` and use `@wordpress/element` only. The `@wordpress/element` is a very thin abstraction layer on top of `react`. Most of it simply re-exports what's there in React. And since most of the WP internal code uses it, it's probably good idea for us to use it as well.

Some libraries might require `react` as a peer dependency. To handle those we can link `react` to `@wordpress/element` in `package.json`. Like we do for `prettier`.

Reply:
:thinking: 

I originally kept react [only because of this import](https://github.com/Automattic/sensei/blob/50679ad675f5a5c5862dcc3c4d577de3299473e0/assets/blocks/quiz/question-block/question-context.js#L4), but [looking here `@wordpress/element` also exports `createContext`](https://github.com/WordPress/gutenberg/blob/b22ce95464d653400954b219e23a8d42285caa9c/packages/element/src/react.js#L70-L77) so..yeah, nice idea. It will help to "remove" one dependency. :D 

About this:

> Some libraries might require react as a peer dependency. To handle those we can link react to @wordpress/element in package.json. Like we do for prettier.

About this... I'm considering using [npm v7's `overrides`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) to link between `react` and `@wordpress/element`, as just changing it on dependencies will not change things for our dependencies and maybe sub-dependencies. 

As a related question, I saw that we already use [yarn's `resolutions` here](https://github.com/Automattic/sensei/blob/3921d82867bc76b88f56c7699239693df228d962/package.json#L96-L100), which I think could also be a good candidate to be removed, as it's ignored by npm and I don't think anyone uses yarn on the team anyway. Another option would be to just rename the `resolutions` field to `overrides`, too.

What do you think?

Reply:
> I don't think that we should worry about dependencies of our dependencies. If we don't like the way a module manages it's dependencies, we shouldn't use that module in the first place.

Okay. Good point. For some reason I read your comment as referring to dependencies with dependency on react, and not with a peer dependency on react, so that confused me a little bit. Sorry for that!

> Not sure about the resolutions part in package.json. I don't know if it's used anywhere.

I tested it here. And it's not used anywhere by npm. The only package manager that's affected by it is yarn, and as we don't use yarn I decided to remove it anyway. This change is on commit [9edb156](https://github.com/Automattic/sensei/pull/4820/commits/9edb15654c038942f77469959df505cfef22000c).

> Some libraries might require react as a peer dependency. To handle those we can link react to @wordpress/element in package.json. Like we do for prettier.

About this, I tried to do that and...our test suite broke. More specifically, some dependencies like `react-with-styles` aren't able to import `@wordpress/element` in the same way they import `react`. Looking further, it looks like the CommonJS build (which is used by `react-with-styles`) of `@wordpress/element` defines the property `__esModule`, but doesn't put everything in the `default` property, which crashes the code as it tries to read from `default` as `__esModule` is true.

Another issue is that `@testing-library/react` tries to import `react-dom`, which ends up using the property `ReactCurrentDispatcher` from `@wordpress/element`, which isn't defined, so it just crashes the test.

In the end, I think just leaving `react` as it is is better than trying to make that alias right now. But I definitely understand the reasoning behind wanting to do that change.

Reply:
Ok, I moved react to devDependencies in [67a3c8f](https://github.com/Automattic/sensei/pull/4820/commits/67a3c8fe391bbdc2ea5cae0dc3ae0c23585cf381) and already fixed our code to use `@wordpress/element` instead of `react` in [980b7ce](https://github.com/Automattic/sensei/pull/4820/commits/980b7ced582c9aedfaa549e3770574f4b6410b62).

I also tried to configure ESLint in [468ae0a](https://github.com/Automattic/sensei/pull/4820/commits/468ae0a6e2be8d30b409bc73e1910d8c74db8598) to properly detect such dependencies and emit an error when someone uses react, for instance, but for some reason, it's not working here, as apparently react is being detected as a "builtin" module and so eslint-plugin-import just ignores it.

Reply:
> resolutions

It seems at some point we had the dependency [`npm-force-resolutions`](https://www.npmjs.com/package/npm-force-resolutions) for that. I remember some weird issues where some packages weren't found, or they changed the package-lock every time, and I think it was to solve that. But it was some time ago. Probably already solved after the npm updates. 🙌

Reply:
Cool! Nowdays we have [npm's override](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) feature if we want to fix something like that again, but as you said it doesn't look a problem for now. :D